Simplified/SignInLogic/OELoginChoiceViewController.swift

```diff
@@ -49,34 +57,42 @@ class OELoginChoiceViewController : UIViewController {
   }
   
 
-  // MARK: -
+  // MARK: - Actions
   
-  private func loginCompletionHandler() {
+  @IBAction func didSelectClever() {
+    didSelectAuthenticationMethod(.clever)
+  }
+
+  @IBAction func didSelectFirstBook() {
+    guard let libraryAccount = AccountsManager.shared.currentAccount else {
+      displayGenericAlert(for: "Unable to get library Account instance after selecting First Book as login choice")
+      return
+    }
+
+    let firstBookVC = OELoginFirstBookVC(libraryAccount: libraryAccount,
+                                         loginSuccessCompletion: completeLogin)
+    navigationController?.pushViewController(firstBookVC, animated: true)
+  }
+  // MARK: - Private
+
+  private func completeLogin() {
     view.window?.rootViewController?.dismiss(animated: true, completion: nil)
     dismiss(animated: true, completion: nil)
-    
-    NYPLSettings.shared.userHasSeenWelcomeScreen = true
+
     guard let appDelegate = UIApplication.shared.delegate else {
-      Log.error("", "Could not load app delegate")
+      displayGenericAlert(for: "Could not load app delegate")
       return
     }
-    
+
     guard let appWindow = appDelegate.window else {
-      Log.error("", "Could not load app window")
+      displayGenericAlert(for: "Could not load app window")
       return
     }
+
     Log.info(#function, "Installing main root VC")
     appWindow?.rootViewController = NYPLRootTabBarController.shared()
   }
 
-  @IBAction func didSelectClever() {
-    didSelectAuthenticationMethod(.clever)
-  }
-
-  @IBAction func didSelectFirstBook() {
-    didSelectAuthenticationMethod(.firstBook)
-  }
-
   private func didSelectAuthenticationMethod(_ loginChoice: LoginChoice) {
```

Comment:
Maybe the body of this function can be moved into `didSelectClever()`? Or make this a `clever` specific method? Since `Firstbook` no longer uses the logic in this method.

Reply:
that's true, but I ignored it because I will need to work on the Clever side soon too, so I  just left it there.